test/Superpower.Tests/Parsers/SpanTests.cs

```diff
@@ -83,5 +84,33 @@ public void RegexMatches()
             var r = parser(input);
             Assert.Equal("Foo", r.Value.ToStringValue());
         }
+
+        [Theory]
+        [InlineData("", "")]
+        [InlineData("STOP", "")]
+        [InlineData("123STOP", "123")]
+        [InlineData("STOP123", "")]
+        [InlineData("123STOP123STOP123", "123")]
+        public void UntilMatches(string text, string expected)
+        {
+            var result = Span.Until("STOP").OptionalOrDefault(TextSpan.Empty).Parse(text);
+            Assert.Equal(expected, result.ToStringValue());
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("123")]
+        [InlineData("12345")]
+        public void UntilParseFails(string text)
+        {
+            Assert.Throws<ParseException>(() => Span.Until("STOP").Parse(text));
```

Comment:
Good to check that the error message and position are what we expect, if possible.

Reply:
Added